File: Herança - Subclasses de Personagem/aula4_heranca_JAVA/Guerreiro.java
package aula4_heranca_JAVA;

public class Guerreiro extends Personagem{

        private int defesa;

public Guerreiro(String nome, int vida, int nivel){
    super(nome, vida, nivel);
    defesa = 5;
}
public int getDefesa(){
    return defesa;
}
public void setDefesa(int defesa){
    if (defesa >= 0){
        this.defesa = defesa;
    } else {
        System.err.println("Defesa inválida");
    }
}
@Override
public void receberDano(int dano){
    super.receberDano(Math.max(0, dano - defesa));
}
@Override
public String ficha(){
    return super.ficha() + "\nDefesa: " + defesa;
}
}

File: Polimorfismo - Batalha Final/python/guerreiro.py
from personagem import Personagem

class Guerreiro(Personagem):
    def __init__(self, nome, vida, nivel):
        super().__init__(nome, vida, nivel)
        self.defesa = 5

    def get_defesa(self):
        return self.defesa

    def set_defesa(self, defesa):
        if defesa >= 0:
            self.defesa = defesa
        else:
            print("Defesa inválida")

    def receber_dano(self, dano):
        super().receber_dano(max(0, dano - self.defesa))

    def ficha(self):
        return super().ficha() + f"\nDefesa: {self.defesa}"

    def habilidade(self):
        return "escudo de aço"